book/07-git-tools/sections/submodules.asc

```diff
@@ -372,6 +375,67 @@ Submodule DbConnector c3f01dc..c87d55d:
 
 Git will by default try to update *all* of your submodules when you run `git submodule update --remote` so if you have a lot of them, you may want to pass the name of just the submodule you want to try to update.
 
+===== Pulling Upstream Changes from the Project Remote
+Let's now step into the shoes of your collaborator, who has his own local clone of the the MainProject repository.
+Simply executing `git pull` to get your newly committed changes is not enough:
+
+[source,console]
+----
+$ git pull
+From https://github.com/chaconinc/MainProject
+   fb9093c..0a24cfc  master     -> origin/master
+Fetching submodule DbConnector
+From https://github.com/chaconinc/DbConnector
+   c3f01dc..c87d55d  stable     -> origin/stable
+Updating fb9093c..0a24cfc
+Fast-forward
+ .gitmodules         | 2 +-
+ DbConnector         | 2 +-
+ 2 files changed, 2 insertions(+), 2 deletions(-)
+
+$ git status
+ On branch master
+Your branch is up-to-date with 'origin/master'.
+Changes not staged for commit:
+  (use "git add <file>..." to update what will be committed)
+  (use "git checkout -- <file>..." to discard changes in working directory)
+
+	modified:   DbConnector (new commits)
+
+Submodules changed but not updated:
+
+* DbConnector c87d55d...c3f01dc (4):
+  < catch non-null terminated lines
+  < more robust error handling
+  < more efficient db routine
+  < better connection routine
+
+no changes added to commit (use "git add" and/or "git commit -a")
+----
+
+By default, the  `git pull` command recursively fetches submodules changes, as we can see in the output of the first command above.
+However, it does not *update* the submodules.
+This is shown by the output of the `git status` command, which shows the submodule is ``modified'', and has ``new commits''.
+What's more, the brackets showing the new commits point left (<), indicating that these commits are recorded in MainProject but are not present in the local DbConnector checkout.
+To finalize the update, you need to run `git submodule update`:
+
+[source,console]
+----
+$ git submodule update --init --recursive
+Submodule path 'vendor/plugins/demo': checked out '48679c6302815f6c76f1fe30625d795d9e55fc56'
+
+$ git status
+ On branch master
+Your branch is up-to-date with 'origin/master'.
+nothing to commit, working tree clean
+----
+
+Note that to be on the safe side, you should run `git submodule update` with the `--init` flag in case the MainProject commits you just pulled added new submodules, and with the `--recursive` flag if any submodules have nested submodules.
+
+If you want to automate this process, you can add the `--recurse-submodules` flag to the `git pull` command (since Git 2.14). This will make Git run `git submodule update` right after the pull, putting the submodules in the correct state.
```

Comment:
One nit: can you make sure all the sentences are on their own line? I see you did this in most places, I think this might be the only one that got missed.

Reply:
Thanks for the quick review! I updated my PR with this correction and a one or two typos corrections in my commit messages.